subsys/bluetooth/mesh/provisioner.c

```diff
@@ -0,0 +1,625 @@
+/*  Bluetooth Mesh */
+
+/*
+ * Copyright (c) 2017 Intel Corporatio
+ * Copyright (c) 2020 Lingao Meng
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <errno.h>
+#include <sys/atomic.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+
+#include <net/buf.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/mesh.h>
+#include <bluetooth/uuid.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_PROVISIONER)
+#define LOG_MODULE_NAME bt_mesh_provisioner
+#include "common/log.h"
+
+#include "host/ecc.h"
+#include "host/testing.h"
+
+#include "crypto.h"
+#include "adv.h"
+#include "mesh.h"
+#include "net.h"
+#include "beacon.h"
+#include "access.h"
+#include "foundation.h"
+#include "proxy.h"
+#include "prov_common.h"
+#include "settings.h"
+#include "prov_bearer.h"
+
+#if defined(CONFIG_BT_MESH_PROVISIONER)
+#define PROVISIONER_LINK 1
+#else
+#define PROVISIONER_LINK 0
+#endif
+
+struct provisioner_link {
+	struct bt_mesh_cdb_node *node;
+	uint16_t addr;
+	uint16_t net_idx;
+	uint8_t attention_duration;
+	uint8_t uuid[16];
+};
+
+struct provisioner_link provisioner[PROVISIONER_LINK];
```

Comment:
Since the file `provisioner.c` is only build when `CONFIG_BT_MESH_PROVISIONER` is set, this whole "trick" is not needed. I would suggest just doing

```
struct provisioner_link {
	struct bt_mesh_cdb_node *node;
	uint16_t addr;
	uint16_t net_idx;
	uint8_t attention_duration;
	uint8_t uuid[16];
} provisioner;
```

Reply:
Great